cores/arduino/stm32/LowPower.c

```diff
@@ -0,0 +1,327 @@
+/**
+  ******************************************************************************
+  * @file    LowPower.c
+  * @author  WI6LABS
+  * @version V1.0.0
+  * @date    17 - November -2017
+  * @brief   Provides a Low Power interface
+  *
+  ******************************************************************************
+  * @attention
+  *
+  * <h2><center>&copy; COPYRIGHT(c) 2017 STMicroelectronics</center></h2>
+  *
+  * Redistribution and use in source and binary forms, with or without modification,
+  * are permitted provided that the following conditions are met:
+  *   1. Redistributions of source code must retain the above copyright notice,
+  *      this list of conditions and the following disclaimer.
+  *   2. Redistributions in binary form must reproduce the above copyright notice,
+  *      this list of conditions and the following disclaimer in the documentation
+  *      and/or other materials provided with the distribution.
+  *   3. Neither the name of STMicroelectronics nor the names of its contributors
+  *      may be used to endorse or promote products derived from this software
+  *      without specific prior written permission.
+  *
+  * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+  * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+  * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+  * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+  * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+  * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+  * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+  * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+  * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+  *
+  ******************************************************************************
+  */
+
+#include "Arduino.h"
+#include "LowPower.h"
+
+#ifdef HAL_PWR_MODULE_ENABLED
+
+#ifdef __cplusplus
+ extern "C" {
+#endif
+
+// NOTE: is these pins are common to a MCU family?
```

Comment:
Hereafter the list of the 21 possibles wake up pins across all STM32 MCU (Fx/H7/Lx)
#define SYS_WKUP0 PA_0
#define SYS_WKUP1 PA_0
#define SYS_WKUP1 PA_2
#define SYS_WKUP1 PC_13
#define SYS_WKUP2 PA_2
#define SYS_WKUP2 PC_0
#define SYS_WKUP2 PC_13
#define SYS_WKUP3 PA_2
#define SYS_WKUP3 PC_1
#define SYS_WKUP3 PE_6
#define SYS_WKUP3 PI_8
#define SYS_WKUP4 PA_2
#define SYS_WKUP4 PC_13
#define SYS_WKUP4 PI_11
#define SYS_WKUP5 PC_1
#define SYS_WKUP5 PC_5
#define SYS_WKUP5 PI_8
#define SYS_WKUP6 PB_5
#define SYS_WKUP6 PI_11
#define SYS_WKUP7 PB_15
#define SYS_WKUP8 PF_2
I will add defines linked to the targeted MCU in the variant to avoid defining a pinmap array and use those define here.

Reply:
It's OK for me